src/System.Data.SqlClient/tests/ManualTests/SQL/UdtTest/UdtTest2.cs

```diff
@@ -342,6 +342,66 @@ public void UDT_DataSetFill()
             }
         }
 
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.IsUdtTestDatabasePresent), nameof(DataTestUtility.AreConnStringsSetup))]
+        public void UDTParams_DeriveParameters()
```

Comment:
I think we also need a test case where the user sets the typename(might be an existing workaround) after the he/she gets it from deriveparameter().

Reply:
Since this is trying ot fix an issue and that correct behaviour is for it to work it might be best to test it by specifically setting an incorrect typename after the derivation step and then making sure it fails as expected. Does that sound reasonable?

Reply:
Yeah, Sounds good to me.

Reply:
I've added this test.